crates/hotshot/types/src/data.rs

```diff
@@ -797,7 +798,80 @@ pub struct QuorumProposal2<TYPES: NodeType> {
 
     /// The light client state update certificate for the next epoch.
     /// This is required for the epoch root.
-    pub state_cert: Option<LightClientStateUpdateCertificate<TYPES>>,
+    pub state_cert: Option<LightClientStateUpdateCertificateV2<TYPES>>,
+}
+
+/// Legacy version of `QuorumProposal2` corresponding to consensus protocol version V3.
+///
+/// `QuorumProposal2` state_cert field was updated to use new
+/// `LightClientStateUpdateCertificateV2`.  
```

Comment:
So I'm kind of struggling to keep all the information about what V{n} means in different contexts. If something is coupled one to one with a consensus protocol version then should we just use the matching version number from there instead? Not sure it would be better.

Also I feel like we need to start keeping more documentation around on which version of types work with what.

This is just a comment, not asking for explicit changes. 😅

Reply:
yeah thats what I did before, but it got more confusing because we have `QuorumProposal2` which is the new version.